modules/EnsEMBL/Web/Object/Gene.pm

```diff
@@ -770,7 +771,24 @@ sub get_homologies {
   return unless defined $query_member;
   
   my $homology_adaptor = $database->get_HomologyAdaptor;
-  my $homologies_array = $homology_adaptor->fetch_all_by_Member($query_member); # It is faster to get all the Homologues and discard undesired entries than to do fetch_all_by_Member_method_link_type
+  # It is faster to get all the Homologues and discard undesired entries than to do fetch_all_by_Member_method_link_type
+  my $homologies_array = [grep { $_->description =~ /$homology_description/ } @{$homology_adaptor->fetch_all_by_Member($query_member)}];
+
+  if ($restrict_to_gene_tree) {
+    my $hub = $self->hub;
+    my $strain_tree = $hub->species_defs->get_config($hub->species,'RELATED_TAXON') if($hub->param('data_action') =~ /strain_/i);
```

Comment:
Please correct me if I am wrong, for metazoa orthologs, `data_action` is `Compara_Ortholog`, does this mean it will not respect protostomes and insects?

Reply:
Metazoa trees aren't represented on the site as strain tree pages (`data_action` won't start with `strain`); so `$strain_tree` will remain undefined. I do not know what consequence this is going to have for this code though.

Reply:
For what it's worth, my intention with setting the `$strain_tree` in this way was to mirror how things are done in `get_homologue_alignments` ([line 873](https://github.com/Ensembl/ensembl-webcode/blob/20fa36a2dca860cee91ee91ba40313c4d7a8418b/modules/EnsEMBL/Web/Object/Gene.pm#L873)), so that the set of homologies returned for OrthoXML/PhyloXML formats with the `$restrict_to_gene_tree` option would match the set of aligned homologue sequences which could be downloaded for other formats. Should I add a comment here explaining this intention?

In any case, `$strain_tree` is indeed expected to be undefined for Metazoa homologies, so IIUC the `clusterset_id` parameter should still be picked up for protostomes and insects in `get_GeneTree` ([here](https://github.com/Ensembl/ensembl-webcode/blob/20fa36a2dca860cee91ee91ba40313c4d7a8418b/modules/EnsEMBL/Web/Object/Gene.pm#L944)).